jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplWithoutAuditTest.java

```diff
@@ -0,0 +1,390 @@
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.drools.compiler.kie.builder.impl.InternalKieModule;
+import org.jbpm.kie.services.impl.KModuleDeploymentUnit;
+import org.jbpm.kie.test.util.AbstractKieServicesBaseTest;
+import org.jbpm.runtime.manager.impl.deploy.DeploymentDescriptorImpl;
+import org.jbpm.services.api.ProcessInstanceNotFoundException;
+import org.jbpm.services.api.model.DeploymentUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.api.builder.ReleaseId;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.internal.KieInternalServices;
+import org.kie.internal.process.CorrelationKey;
+import org.kie.internal.runtime.conf.AuditMode;
+import org.kie.internal.runtime.conf.DeploymentDescriptor;
+import org.kie.internal.runtime.conf.DeploymentDescriptorBuilder;
+import org.kie.scanner.KieMavenRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProcessServiceImplWithoutAuditTest extends AbstractKieServicesBaseTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProcessServiceImplWithoutAuditTest.class);
+
+    private List<DeploymentUnit> units = new ArrayList<DeploymentUnit>();
+
+    @Before
+    public void prepare() {
+    	configureServices();
+    	logger.debug("Preparing kjar");
+        KieServices ks = KieServices.Factory.get();
+        ReleaseId releaseId = ks.newReleaseId(GROUP_ID, ARTIFACT_ID, VERSION);
+        List<String> processes = new ArrayList<String>();
+        processes.add("repo/processes/general/customtask.bpmn");
+        processes.add("repo/processes/general/humanTask.bpmn");
+        processes.add("repo/processes/general/import.bpmn");
+        processes.add("repo/processes/general/signal.bpmn");
+		processes.add("repo/processes/general/signalWithExpression.bpmn2");
+        processes.add("repo/processes/general/callactivity.bpmn");
+        
+        DeploymentDescriptor customDescriptor = new DeploymentDescriptorImpl("org.jbpm.domain");
+        DeploymentDescriptorBuilder ddBuilder = customDescriptor.getBuilder();
+        ddBuilder.auditMode(AuditMode.NONE);        
+        Map<String, String> extraResources = new HashMap<String, String>();        
+        extraResources.put("src/main/resources/" + DeploymentDescriptor.META_INF_LOCATION, customDescriptor.toXml());
+
+        InternalKieModule kJar1 = createKieJar(ks, releaseId, processes, extraResources);
+        File pom = new File("target/kmodule", "pom.xml");
+        pom.getParentFile().mkdir();
+        try {
+            FileOutputStream fs = new FileOutputStream(pom);
+            fs.write(getPom(releaseId).getBytes());
+            fs.close();
+        } catch (Exception e) {
+
```

Comment:
When exception is thrown at this point, the test setup might be compromised. Please throw an unchecked exception to make it clear what the root case is.

Reply:
added throw or unchecked exception